src/App.tsx

```diff
@@ -1,56 +1,69 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+import { SearchInput } from './components/SearchInput/SearchInput';
+import { filterPeopleByQuery, queryDebounce } from './utils';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [queryValue, setQueryValue] = useState('');
+  const [queryApplied, setQueryApplied] = useState('');
+  const [personSelected, setPersonSelected] = useState(peopleFromServer[0]);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const { name, born, died } = personSelected;
+
+  const applyQuery = useCallback(
+    queryDebounce((query) => {
+      setQueryApplied(query);
+      setIsDropdownVisible(true);
+    }, 1000),
+    [],
+  );
+
+  const handleInputChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setIsDropdownVisible(false);
+      setQueryValue(event.target.value);
+      applyQuery(event.target.value);
+    },
+    [],
+  );
+
+  const handleInputFocus = useCallback(() => {
+    setIsDropdownVisible(true);
+  }, []);
+
+  const handlePersonSelect = useCallback((person: Person) => {
+    setIsDropdownVisible(false);
+    setPersonSelected(person);
+    setQueryValue(person.name);
+    setQueryApplied(person.name);
+  }, []);
```

Comment:
Looks like you are over-using `useCallback` hooks

When used in the wrong way they hurt the performance

In your particular case these 4 hoks have no effect